pkg/controller/provider/powerdns/factory.go

```diff
@@ -38,7 +38,7 @@ func newAdapter() provider.DNSHandlerAdapter {
 	checks.Add(provider.RequiredProperty("Server", "server").
 		Validators(provider.NoTrailingWhitespaceValidator, provider.PrintableValidator, provider.URLValidator("https", "http"), provider.MaxLengthValidator(256)))
 	checks.Add(provider.RequiredProperty("ApiKey", "apiKey").
-		Validators(provider.NoTrailingWhitespaceValidator, provider.PrintableValidator, provider.MaxLengthValidator(128)).
+		Validators(provider.NoTrailingWhitespaceValidator, provider.PrintableValidator, provider.MaxLengthValidator(8192)).
```

Comment:
nit: Could you add a comment explaining why we're using `8192` as the maximum length?
```suggestion
		Validators(provider.NoTrailingWhitespaceValidator, provider.PrintableValidator, provider.MaxLengthValidator(8192)). // PowerDNS does not impose a maximum length for API keys. Therefore, the typical maximum length of HTTP headers is used.
```